tools/tinytest-codegen.py

```diff
@@ -32,6 +32,35 @@ def script_to_map(test_file):
     return r
 
 
+def load_profile(profile_file):
```

Comment:
Rather than writing a custom parser, what if the profiles were just Python code

```
def load_profile(profile_file):
  profile_globals = {"test_dirs": set(), "exclude_tests: set() }
  exec(profile_file, profile_globals)
  return profile_globals["test_dirs"], profile_globals["exclude_tests"]
```

Reply:
This would allow you to write either:

```
test_dirs.add(...)
```

or

```
test_dirs = ...
```

in the profile script.

Reply:
I'm just not really fond of `exec` myself, but your approach in this case makes more sense if `test_dirs` and `exclude_tests` are pre-populated with the default lists.  This way one may not only add new entries but delete ones that aren't relevant for a particular port (or add conditional exclusions if some environment variables set by `make` are passed down to the configuration script, like the board name for instance).

Reply:
Also, if the global `test_dirs` and `exclude_tests` variables are filtered through the configuration script rather than composed from the configuration script output it also resolves the nomenclature remark.  The global variables get reassigned from the output of `load_profile` so no new variables are defined.

Reply:
Ok, I've switched to a Python script as a configuration file. I wonder whether I should change the configuration script extension to `.py` and let `ruff` skip error F821 on that file as well in `pyproject.toml` (or maybe use `ports/**/tests_profile.py`?)

Reply:
I think it's fine to leave it as a `.txt`.